pulpcore/cli/ansible/distribution.py

```diff
@@ -0,0 +1,113 @@
+from typing import Optional, Dict, Any
+
+import click
+
+
+from pulpcore.cli.common.generic import (
+    list_entities,
+    show_by_name,
+    destroy_by_name,
+)
+from pulpcore.cli.common.context import (
+    PulpContext,
+    pass_pulp_context,
+    pass_entity_context,
+)
+from pulpcore.cli.ansible.context import (
+    PulpAnsibleDistributionContext,
+    PulpAnsibleRepositoryContext,
+)
+
+
+@click.group()
+@click.option(
+    "-t",
+    "--type",
+    "distribution_type",
+    type=click.Choice(["ansible"], case_sensitive=False),
+    default="ansible",
+)
+@pass_pulp_context
+@click.pass_context
+def distribution(ctx: click.Context, pulp_ctx: PulpContext, distribution_type: str) -> None:
+    if distribution_type == "ansible":
+        ctx.obj = PulpAnsibleDistributionContext(pulp_ctx)
+    else:
+        raise NotImplementedError()
+
+
+distribution.add_command(list_entities)
+distribution.add_command(show_by_name)
+distribution.add_command(destroy_by_name)
+
+
+# TODO Add content_guard option
+@distribution.command()
+@click.option("--name", required=True)
+@click.option("--base-path", required=True)
+@click.option("--repository", help="name of repository")
+@click.option("--version", help="version of repository, leave blank for always latest")
+@pass_entity_context
+@pass_pulp_context
+def create(
+    pulp_ctx: PulpContext,
+    distribution_ctx: PulpAnsibleDistributionContext,
+    name: str,
+    base_path: str,
+    repository: Optional[str],
+    version: Optional[str],
+) -> None:
+    body: Dict[str, Any] = {"name": name, "base_path": base_path}
+    repo: Dict[str, Any] = PulpAnsibleRepositoryContext(pulp_ctx).find(name=repository)
+    if version:
+        body["repository_version"] = f'{repo["versions_href"]}{version}/'
+    elif repository:
+        body["repository"] = repo["pulp_href"]
```

Comment:
Do you want me to add this logic to other plugin commands in a separate pr?

Reply:
That would be awesome. And yes, please make it a separate PR. It is easier to review, because a smaller PR is less code to look at and usually needs less iterations.

My estimation is: total_review_time is in O(number_of_lines²)